scripts/update-contributors.js

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env node
+
+/**
+ * This script updates the "contributors" property of the root `package.json`.
+ * It modifies `package.json` in place!
+ *
+ * See `.mailmap` for username/email mappings.
+ */
+
+'use strict';
+
+const exec = require('child_process').exec;
+const path = require('path');
+const fs = require('fs');
+
+const ROOT = path.join(__dirname, '..');
+
+// list of authors/emails that should not appear in the contributors list, e.g. bots
+const BLACKLIST = [
+  'greenkeeperio-bot <support@greenkeeper.io>',
+  'greenkeeper[bot] <greenkeeper[bot]@users.noreply.github.com>',
+  'TJ Holowaychuk <tj@vision-media.ca>' // author
+];
+
+// could use `| sort | uniq` here but didn't want to assume 'nix
+exec(
+  'git log --format="%aN <%aE>"',
+  {
+    cwd: ROOT
+  },
+  (err, res) => {
+    if (err) {
+      throw err;
+    }
+    const contributors = Array.from(new Set(res.trim().split(/\r?\n/)))
+      .filter(contributor => BLACKLIST.indexOf(contributor) < 0)
+      .sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase()));
+    const newCount = contributors.length;
+    const pkgFilepath = path.join(ROOT, 'package.json');
+    const pkg = JSON.parse(fs.readFileSync(pkgFilepath, 'utf8'));
```

Comment:
How about you just do `require(pkgFilepath);`?

Reply:
In this case, it’s mainly semantic. If I don’t need nor want the caching which `require()` provides, then I shouldn’t use it. In this script it doesn’t really matter since we don’t use the result elsewhere, but I feel it better illustrates the intent.

Reply:
I understood what you mean. :)